internal/providers/msentraid/msentraid.go

```diff
@@ -86,14 +90,34 @@ func (p Provider) GetExtraFields(token *oauth2.Token) map[string]interface{} {
 	}
 }
 
+// GetMetadata returns relevant metadata about the provider.
+func (p Provider) GetMetadata(provider *oidc.Provider) (map[string]interface{}, error) {
+	var claims struct {
+		MSGraphHost string `json:"msgraph_host"`
+	}
+
+	if err := provider.Claims(&claims); err != nil {
+		return nil, fmt.Errorf("failed to get provider claims: %v", err)
+	}
+
+	return map[string]interface{}{
+		"msgraph_host": claims.MSGraphHost,
+	}, nil
+}
+
 // GetUserInfo is a no-op when no specific provider is in use.
-func (p Provider) GetUserInfo(ctx context.Context, accessToken *oauth2.Token, idToken *oidc.IDToken) (info.User, error) {
+func (p Provider) GetUserInfo(ctx context.Context, accessToken *oauth2.Token, idToken *oidc.IDToken, providerMetadata map[string]interface{}) (info.User, error) {
```

Comment:
Now that we are touching the function signature, maybe we should sneak in an update to the function comment? We missed updating it when adding this code to the `msentraid` package and it deserves a better comment.

Reply:
good point, done